partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -1725,14 +1725,30 @@ internal class PartiQLParserDefault : PartiQLParser {
         }
 
         override fun visitFunctionCall(ctx: GeneratedParser.FunctionCallContext) = translate(ctx) {
-            val function = visit(ctx.functionName()) as Identifier
             val args = visitOrEmpty<Expr>(ctx.expr())
-            exprCall(function, args)
+            when (val funcName = ctx.functionName()) {
+                is GeneratedParser.FunctionNameReservedContext -> {
+                    when (funcName.name.type) {
+                        GeneratedParser.MOD -> exprBinary(Expr.Binary.Op.MODULO, args[0], args[1])
+                        else -> visitNonReservedFunctionCall(ctx, args)
```

Comment:
I'm very curious to explore why we have the reserved function names at all. Other SQL ANTLR grammars simply use a qualified name. I don't think we should change it here, but let's check with John as to why he included those.

Reply:
Agree. Maybe we can have an alias parameter in FnSignature or something to resolve the multiple naming issue. (i.e., `char_length` and `character_length`)